pkg/util/openstack/network.go

```diff
@@ -101,24 +102,41 @@ func GetFloatingNetworkID(client *gophercloud.ServiceClient) (string, error) {
 	if err != nil {
 		return "", mc.ObserveRequest(err)
 	}
+
 	for _, externalNet := range externalNetworks {
```

Comment:
Hmm, not sure why the variable is called `externalNetworks` because it looks to me like it contains _all_ networks, regardless of whether they're external or not.

We do not filter anything in on line 96: https://github.com/kubernetes/cloud-provider-openstack/pull/2014/files#diff-a89aa055a2316c0db33d258ec1c17fc3948b8f53fbc3a475d5a6602025096b33R96

Reply:
Right. In [gophercloud](https://github.com/gophercloud/gophercloud/blob/master/openstack/networking/v2/extensions/external/doc.go#L19) it's documented as you mentioned. Let me fix that.